### ISSUE TITLE
[tune/ray] Error messages are too explosive

### ISSUE THREAD

**Issue by himat** [ray-project/ray#5124]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **Ray installed from (source or binary)**: binary
- **Ray version**: 0.7.1
- **Python version**: 3.6.5

### Describe the problem

Is it the expected behavior that Ray crashes when I do ctrl+C?

### Source code / logs

```

(pid=4218) [INFO]: del - Epoch 1
(pid=4218)
(pid=4218) [INFO]: lr: [0.007]
(pid=4218)
(pid=4218) [train] Epoch 1:   0%|          | 0/17 [00:00<?, ?it/s]
2019-07-05 16:26:45,004 WARNING util.py:64 -- The `experiment_checkpoint` operation took 0.17577290534973145 seconds to complete, which may be a performance bottleneck.
^CTraceback (most recent call last):
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/ray/worker.py", line 2316, in wait
    worker.current_task_id,
  File "python/ray/_raylet.pyx", line 269, in ray._raylet.RayletClient.wait
  File "python/ray/_raylet.pyx", line 60, in ray._raylet.check_status
Exception: [RayletClient] Raylet connection closed.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "main.py", line 151, in <module>
    main()
  File "main.py", line 77, in main
    run_tune(train, config)
  File "main.py", line 140, in run_tune
    "weight_decay": tune.grid_search([0.0001, 0.05, 0.1, 0.2]),
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/ray/tune/tune.py", line 254, in run
    runner.step()
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/ray/tune/trial_runner.py", line 266, in step
    self._process_events()  # blocking
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/ray/tune/trial_runner.py", line 430, in _process_events
    trial = self.trial_executor.get_next_available_trial()  # blocking
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/ray/tune/ray_trial_executor.py", line 299, in get_next_available_trial
    [result_id], _ = ray.wait(shuffled_results)
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/ray/worker.py", line 2316, in wait
    worker.current_task_id,
KeyboardInterrupt
Exception in thread ray_import_thread:
Traceback (most recent call last):
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/connection.py", line 185, in _read_from_socket
    raise socket.error(SERVER_CLOSED_CONNECTION_ERROR)
OSError: Connection closed by server.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/ray/import_thread.py", line 69, in _run
    msg = import_pubsub_client.get_message()
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/client.py", line 3135, in get_message
    response = self.parse_response(block=False, timeout=timeout)
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/client.py", line 3036, in parse_response
    return self._execute(connection, connection.read_response)
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/client.py", line 3013, in _execute
    return command(*args)
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/connection.py", line 637, in read_response
    response = self._parser.read_response()
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/connection.py", line 290, in read_response
    response = self._buffer.readline()
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/connection.py", line 224, in readline
    self._read_from_socket()
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/connection.py", line 199, in _read_from_socket
    (e.args,))
redis.exceptions.ConnectionError: Error while reading from socket: ('Connection closed by server.',)

Exception in thread ray_listen_error_messages:
Traceback (most recent call last):
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/connection.py", line 185, in _read_from_socket
    raise socket.error(SERVER_CLOSED_CONNECTION_ERROR)
OSError: Connection closed by server.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/ray/worker.py", line 1646, in listen_error_messages_raylet
    msg = worker.error_message_pubsub_client.get_message()
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/client.py", line 3135, in get_message
    response = self.parse_response(block=False, timeout=timeout)
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/client.py", line 3036, in parse_response
    return self._execute(connection, connection.read_response)
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/client.py", line 3013, in _execute
    return command(*args)
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/connection.py", line 637, in read_response
    response = self._parser.read_response()
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/connection.py", line 290, in read_response
    response = self._buffer.readline()
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/connection.py", line 224, in readline
    self._read_from_socket()
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/connection.py", line 199, in _read_from_socket
    (e.args,))
redis.exceptions.ConnectionError: Error while reading from socket: ('Connection closed by server.',)

Exception in thread ray_print_logs:
Traceback (most recent call last):
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/connection.py", line 185, in _read_from_socket
    raise socket.error(SERVER_CLOSED_CONNECTION_ERROR)
OSError: Connection closed by server.

During handling of the above exception, another exception occurred:

Exception in thread ray_print_logs:
Traceback (most recent call last):
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/connection.py", line 185, in _read_from_socket
    raise socket.error(SERVER_CLOSED_CONNECTION_ERROR)
OSError: Connection closed by server.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/ray/worker.py", line 1544, in print_logs
    msg = pubsub_client.get_message()
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/client.py", line 3135, in get_message
    response = self.parse_response(block=False, timeout=timeout)
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/client.py", line 3036, in parse_response
    return self._execute(connection, connection.read_response)
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/client.py", line 3013, in _execute
    return command(*args)
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/connection.py", line 637, in read_response
    response = self._parser.read_response()
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/connection.py", line 290, in read_response
    response = self._buffer.readline()
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/connection.py", line 224, in readline
    self._read_from_socket()
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/redis/connection.py", line 199, in _read_from_socket
    (e.args,))
redis.exceptions.ConnectionError: Error while reading from socket: ('Connection closed by server.',)

Exception ignored in: <object repr() failed>
Traceback (most recent call last):
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/ray/actor.py", line 613, in __del__
AttributeError: 'NoneType' object has no attribute 'get_global_worker'
Exception ignored in: <object repr() failed>
Traceback (most recent call last):
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/ray/actor.py", line 613, in __del__
AttributeError: 'NoneType' object has no attribute 'get_global_worker'

```

**Comment by richardliaw** [comment]
Hm, can you try `0.7.2` also?

cc @pcmoritz

**Comment by himat** [comment]
Yes, I got the same result pretty much on 0.7.2.

Is ray supposed to handle user interrupts? Or should I be catching those?

**Comment by richardliaw** [comment]
Ray should be handling these user interrupts and providing cleaner/saner error messages. I can address this soon - in the meantime, this shouldn't affect your program right?

**Comment by himat** [comment]
Right, it doesn't affect me, so it's not a big issue.

**Comment by richardliaw** [comment]
Should be addressed in #5237.
